lib/modules/datasource/helm/index.ts

```diff
@@ -63,9 +62,11 @@ export class HelmDatasource extends Datasource {
       }
       const result: HelmRepositoryData = {};
       for (const [name, releases] of Object.entries(doc.entries)) {
-        const latestRelease = releases.sort((r0, r1) =>
-          gt(r0.version, r1.version) ? -1 : 1
-        )[0];
+        const [latestRelease] = releases
+          .filter(({ version }) => helmVersioning.api.isValid(version))
+          .sort((r0, r1) =>
+            helmVersioning.api.isGreaterThan(r0.version, r1.version) ? -1 : 1
```

Comment:
```suggestion
            helmVersioning.api.sortVersions(r0.version, r1.version)
```
as i mentioned before: use `sortVersions`. it's there for exactly this purpose.